module/VuFind/src/VuFind/RecordDriver/EDS.php

```diff
@@ -660,8 +660,27 @@ protected function toHTML($data, $group = null)
                 '<a id="$1" href="#$2"',
                 $data
             );
-
             $data = $this->replaceBRWithCommas($data, $group);
+
+            // ensure tables & mathML are not shown as raw html
+            if (strpos($data, 'ephtml') > -1) {
+                $pattern = "/<\/?ephtml>/";
+                $splitParts = preg_split($pattern, $data);
+                $data = '';
+                foreach ($splitParts as $part) {
+                    if (strpos($part, 'table') > -1 && strpos($part, 'table') < 6) {
```

Comment:
What is the purpose of the "magic number" 6 here?

Reply:
Great question and a place where I should improve. The data will look like this:

`<ephtml> &lt;table&gt;&lt;tbody valign="top"&gt; ...`

After the preg_split it should be:

` &lt;table&gt;&lt;tbody valign="top"&gt; ...`

I probably should simply look for `&lt;table` and then just accept it if it is > -1. For math I could look for `&lt;math`.

I'll look into adding a test. you have been most helpful, as always!

Reply:
Yes, I think that your proposed change sounds reasonable and should make the code easier to understand. Thanks!

Reply:
@demiankatz I did push this change, sorry it took me so long